src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertPlaceholderToInterpolatedStringRefactoringProvider.cs

```diff
@@ -15,351 +15,303 @@
-            }
 
-            var stringInvocationMethods = CollectMethods(stringType, ImmutableArray.Create(nameof(string.Format)));
-            var compositeFormattedInvocationMethods = s_compositeFormattedMethods
-                .SelectMany(pair => CollectMethods(semanticModel.Compilation.GetTypeByMetadataName(pair.typeName), pair.methods))
-                .ToImmutableArray();
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
 
-            var allInvocationMethods = stringInvocationMethods.AddRange(compositeFormattedInvocationMethods);
+            var (invocation, placeholderArgument) = await TryFindInvocationAsync().ConfigureAwait(false);
+            if (invocation is null || placeholderArgument is null)
+                return;
 
-            if (allInvocationMethods.Length == 0)
-            {
+            var placeholderExpression = syntaxFacts.GetExpressionOfArgument(placeholderArgument);
+            var stringToken = placeholderExpression.GetFirstToken();
+
+            // don't offer if the string argument has errors in it, or if converting it to an interpolated string creates errors.
+            if (stringToken.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
                 return;
-            }
 
-            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
-            if (syntaxFactsService == null)
+            // Not supported if there are any omitted arguments following the placeholder.
+            var arguments = syntaxFacts.GetArgumentsOfInvocationExpression(invocation);
+            var placeholderIndex = arguments.IndexOf(placeholderArgument);
+            Contract.ThrowIfTrue(placeholderIndex < 0);
+            for (var i = placeholderIndex + 1; i < arguments.Count; i++)
             {
-                return;
+                var argument = arguments[i];
+                if (syntaxFacts.GetExpressionOfArgument(argument) is null)
+                    return;
+
+                if (syntaxFacts.GetRefKindOfArgument(argument) != RefKind.None)
+                    return;
             }
 
-            var (invocationSyntax, invocationSymbol) = await TryFindInvocationAsync(textSpan, document, semanticModel, allInvocationMethods, syntaxFactsService, context.CancellationToken).ConfigureAwait(false);
-            if (invocationSyntax is null || invocationSymbol is null)
-            {
+            if (semanticModel.GetSymbolInfo(invocation, cancellationToken).GetAnySymbol() is not IMethodSymbol invocationSymbol)
                 return;
-            }
 
-            if (!IsArgumentListCorrect(syntaxFactsService.GetArgumentsOfInvocationExpression(invocationSyntax), invocationSymbol, allInvocationMethods, semanticModel, syntaxFactsService, cancellationToken))
+            // If the user is actually passing an array to a params argument, we can't change this to be an interpolated string.
+            if (invocationSymbol.IsParams())
```

Comment:
❔  Aren't most uses of this calls to `params` methods?

Reply:
it def applies to cases where people have the pattern of a few overloads with varying arg lengths (like 0-4).